device.go

```diff
@@ -22,13 +22,19 @@ import (
 	"time"
 )
 
+// DevicePath is a representation of a USB port path
+// The path describes the physical port connections, ordered from Bus to Device.
+type DevicePath []int
+
 // DeviceDesc is a representation of a USB device descriptor.
+// Root Hubs are represented solely by Bus, do not have a path, and have a port value of 0.
```

Comment:
this belong in the description of `Port` and `Path`

Reply:
perhaps something like this:

```
// The physical port of the parent hub two which the device is connected. Ports are numbered from 1.
// Each bus has one root hub device, Port of the root hub is always set to 0.
Port int 
// The path of port numbers from the root hub device to this device, in that order.
// A path length of 1 indicates that this device is connected directly to the root hub,
// a path length of 2 means there is one hub between this device and the root hub, etc.
// The root hub device has an empty Path.
Path []int
```

Reply:
I've put a mix of yours plus an example which i hope makes it fully clear.  Ultimately we're being consistent to the descriptors in /sys/bus/usb/devices which seems to be best - although lsusb (usbutils) lists root hubs as port 1 just to confuse matters

Reply:
yeah, that's unfortunate (port 1 for the "bus port"). The USB spec doesn't really specify the meaning of "port" other than for hubs. I suppose "port 1 and only" on the bus is as good as no port as far as user-visible presentation goes.